dao.go

```diff
@@ -23,3 +23,12 @@ func (d *Dao) AddDevice(device *DevicePayload) (int, error) {
 	d.data[dev.Id] = *dev
 	return dev.Id, nil
 }
+
+func (d *Dao) GetDevice(id int) (*Device, error) {
+	for _, device := range d.data {
+		if device.Id == id {
+			return &device, nil
+		}
+	}
+	return nil, ErrNotFound("")
```

Comment:
Usually this is not an error - the object was just not found, its normal situation. Some golang libraries handle this situation with the use of an error but to be honest for me the best solution is to use `nil`.

By specifying return type as `(*Device, error)` the signature of the method says "I can return no device at all and you have to handle it". I would return `nil` here and in handler return 404 in case the returned object is nil.